packages/nimble-components/src/table/components/row/index.ts

```diff
@@ -20,13 +21,32 @@ export class TableRow<
     @observable
     public data?: TData;
 
-    // TODO: Temporarily assume the set of columns will be an array of strings.
-    // Eventually, this will be an array of column definitions.
     @observable
-    public columns: string[] = [];
+    public columns: TableColumn[] = [];
 
-    public getCellValue(column: string): TableFieldValue {
-        return this.data ? this.data[column] : undefined;
+    public getCellValue(column: TableColumn): TableCellState<TableRecord> {
+        const fieldNames = column.getRecordFieldNames();
+        if (this.hasValidFieldNames(fieldNames)) {
+            const cellDataValues = fieldNames.map(field => this.data![field]);
+            const cellData = Object.fromEntries(
+                column.cellStateDataFieldNames.map((k, i) => [
+                    k,
+                    cellDataValues[i]
+                ])
+            );
+            const columnConfig = column.getColumnConfig?.() ?? {};
+            const cellState: TableCellState<TableRecord> = {
+                data: cellData,
+                columnConfig
+            };
+            return cellState;
+        }
+
+        return { data: {}, columnConfig: {} };
```

Comment:
Is this the case that gets hit when a table column expects certain fields to be present in the data but they aren't there? How does that manifest in the table, as empty data? Should we consider adding to the new `validity` API to report that this is the reason the table is blank?

If you think this is a good idea it could happen in a follow up. Or if it's already been discussed or I'm misunderstanding, feel free to reject.

Reply:
I'd like to handle this in a follow-up.